packages/react/src/exchanges/hooks/useCreateExchangeFilter.ts

```diff
@@ -36,27 +51,30 @@ function useCreateExchangeFilter() {
         return Promise.reject(new Error('No data to filter provided'));
       }
 
-      reset();
+      if (!orderItemUuid) {
+        return Promise.reject(new Error('No orderItemUuid provided'));
+      }
 
-      return createExchangeFilter(exchangeFilterData, config);
+      return createExchangeFilter(exchangeFilterData, orderItemUuid, config);
     },
-    [createExchangeFilter, reset],
+    [createExchangeFilter, orderItemUuid],
   );
 
   const data = useMemo(() => {
-    if (!exchangeFilterResult) {
+    if (!exchangeFilters && !exchangeFilterById) {
       return undefined;
     }
 
-    return exchangeFilterResult;
-  }, [exchangeFilterResult]);
+    return { exchangeFilters, exchangeFilterById };
+  }, [exchangeFilters, exchangeFilterById]);
 
   return {
     isCreating,
     error,
     data,
     actions: {
       create,
+      reset,
```

Comment:
The `reset` action also does not make sense to be exported here since it will reset all state and not the one that was resulted from the user invocation of the hook.